### ISSUE TITLE
Brush with BarChart shows different selection than selected with brush range

### ISSUE THREAD

**Issue by 94Conti** [recharts/recharts#4629]
## Do you want to request a feature or report a bug?
I would like to report a bug when setting a BarChart inside a Brush, the visual range selection in Brush does not match the data showed in original BarChart. I found this reported issue https://github.com/recharts/recharts/issues/4193 but seems to still reproduceable for BarChart.

Here the original BarChart with full selection working as expected:
![image](https://github.com/recharts/recharts/assets/16888029/2983e368-dd63-4c71-89d0-f2ad80f590f3)

## What is the current behavior?
As you can see in image below the visual range is on top of Bar 5, but the original BarChart shows 4:
![image](https://github.com/recharts/recharts/assets/16888029/7f224150-5214-4625-819f-2d1b9a59c83e)

And here the visual range is on top of the very end of Bar 6, but BarChart shows 5 and 6:
![image](https://github.com/recharts/recharts/assets/16888029/23372cf3-22d4-4eeb-b60a-969b465bba8d)

**Reproduction link:**
https://codesandbox.io/p/sandbox/brush-bar-chart-6vv4jt

## What is the expected behavior?
I expect the original BarChart data matches the visual range selection in Brush BarChart. So for example, when Brush range is on top of 6 it shows only Bar 6.

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?
I had version 2.10.4 installed and I currently update it to latest version 2.12.7 and it's reproduceable.
Browser is Chrome Version 125.0.6422.142 and OS Windows 11.

Is there a way to handle this or it's a bug and needs to be fixed?

Thank you in advance,
Anna

**Comment by ckifer** [comment]
Hi! This is probably more of an enhancement these things have ever been connected. 

Brush uses indices to determine what is being shown (i.e. start of brush is 0, end of brush is data.length - 1), where those indices change in the brush are the same if you have a panorama chart or not (and always have been). This doesn't change if you add a panorama chart.

It does make sense to connect them when a panorama exists though!

**Comment by stephen-oconnor** [comment]
I think I am having similar issues, although maybe I do not understand how the brush functionality should work.

Let's say we have a line chart, 12 data points, Jan - Dec (Pic 1). User moves the right traveller from Dec to Nov, I expect the data points in the chart to update accordingly, e.g. only display Jan - Nov (Pic 2), all good. User then drags the whole brush, which is currently 11 data points to the far right or end, I expect the data points in the chart to display 11 data points, Feb - Dec (Pic 3) but that's not how it currently works. When dragged to the end all of the data points are displayed. Is there something I am not understanding? How can we display an extra data point in the chart after one has been removed via the brush?

This also happens with smaller amended increments via the brush, e.g. set to 4 months, then drag the whole brush. We sometimes display 4 data points, sometimes 5, sometimes 3 ... I don't understand how it should work.

Images below to aid.

**Pic 1**
<img width="800" alt="Screenshot 2024-06-06 at 14 52 10" src="https://github.com/recharts/recharts/assets/4642868/fa84dc50-823f-4e57-b8c5-081d2fc1b02b">

**Pic 2**
<img width="767" alt="Screenshot 2024-06-06 at 14 52 37" src="https://github.com/recharts/recharts/assets/4642868/753ae48a-97b1-4fe6-8d46-f3ba3df66605">

**Pic 3**
<img width="800" alt="Screenshot 2024-06-06 at 14 52 49" src="https://github.com/recharts/recharts/assets/4642868/83f044c6-5d7c-4206-8d2e-8ed62520782f">

Thanks.

**Comment by ckifer** [comment]
@94Conti no, unless you'd like to contribute! We're working on 3.0 which is mostly much required maintenance that will make features like this easier to add. It could come in after that though.

@stephen-oconnor hi! I agree with your assessment. The way it is working is honestly a mystery to me too at the moment, we don't any of the original implementers working on the project so a lot of knowledge has been lost. It would be awesome if you could re-create your examples in a code sandbox and make a new issue for this! This seems related but not the same as @94Conti's ask

**Comment by stephen-oconnor** [comment]
@ckifer no worries, thanks for the response. I'll create an issue with a sandbox no bother. Thanks.

**Comment by 94Conti** [comment]
Thank you @ckifer, we will be following the updates!
